Dockerfile

```diff
@@ -30,36 +35,37 @@ ENV LDFLAGS=-s
 
 FROM base AS cpu
 # amd64 uses gcc which requires devtoolset-11 for AVX extensions while arm64 uses clang
-RUN if [ "$(uname -m)" = "x86_64" ]; then yum install -y devtoolset-11-gcc devtoolset-11-gcc-c++; fi
-ENV PATH=/opt/rh/devtoolset-11/root/usr/bin:$PATH
+RUN if [ "$(uname -m)" = "x86_64" ]; then dnf install -y gcc-toolset-11-gcc gcc-toolset-11-gcc-c++; fi
+ENV PATH=/opt/rh/gcc-toolset-11/root/usr/bin:$PATH
 RUN --mount=type=cache,target=/root/.ccache \
     cmake --preset 'CPU' \
-        && cmake --build --parallel --preset 'CPU' \
-        && cmake --install build --component CPU --strip --parallel 8
+        && cmake --build --parallel $(expr $(nproc) / 2 + 1 ) --preset 'CPU' \
```

Comment:
I believe `--parallel` already defaults to the number of cores `$(expr $(nproc) / 2 + 1 )` might be overkill and harder to read

Reply:
Especially in the build process of GitHub Actions, you most likely always just have 4 vCPUs (aka 4 threads) to your disposable. With 16GB of RAM.

Reply:
My intent was to balance out the docker build stage parallelism to not overshoot the logical CPU count too much.  For example, on arm64 on a 16 core system, we have 5 parallel stages in the build (cuda v11, cuda v12, jetpack 5, jetpack 6, and cpu) so we'll be launching ~80 compiles in parallel, not just 16.   It does seem like the system can handle it, so I'll remove this, but I worry we may thrash on the github action runners.